pkg/controller/deployment/recreate.go

```diff
@@ -126,7 +141,16 @@ func oldPodsRunning(newRS *apps.ReplicaSet, oldRSs []*apps.ReplicaSet, podMap ma
 }
 
 // scaleUpNewReplicaSetForRecreate scales up new replica set when deployment strategy is "Recreate".
-func (dc *DeploymentController) scaleUpNewReplicaSetForRecreate(ctx context.Context, newRS *apps.ReplicaSet, deployment *apps.Deployment) (bool, error) {
-	scaled, _, err := dc.scaleReplicaSetAndRecordEvent(ctx, newRS, *(deployment.Spec.Replicas), deployment)
+func (dc *DeploymentController) scaleUpNewReplicaSetForRecreate(ctx context.Context, newRS *apps.ReplicaSet, deployment *apps.Deployment) (scaled bool, err error) {
+	newScale := *(deployment.Spec.Replicas)
+	if utilfeature.DefaultFeatureGate.Enabled(features.DeploymentPodReplacementPolicy) {
```

Comment:
I actually had the policy check in this place before, but removed it to avoid code duplication, since the same check is done in the `NewRSNewReplicas` call. Also, the `NewRSNewReplicas` resolves to the same value (`deployment.Spec.Replicas`) if no policy is specified.

I agree, it is better to do this check twice to have clear code paths and better readability. I have updated this place to include the policy check.